src/apm/health.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package apm
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/newrelic/k8s-agents-operator/src/api/v1alpha2"
+)
+
+const (
+	envHealthFleetControlFilepath = "NEW_RELIC_FLEET_CONTROL_HEALTH_PATH"
+	envHealthListenPort           = "NEW_RELIC_SIDECAR_LISTEN_PORT"
+	healthSidecarContainerName    = "newrelic-apm-health"
+	healthVolumeName              = "newrelic-apm-health"
```

Comment:
Do we want both of them to have the same name. May be change the name of the sidecar container to have `newrelic-apm-health-sidecar`?

Reply:
Also needs to be changed here: https://github.com/newrelic/k8s-agents-operator/blob/1e212ab96a852805a8949f37cb0d2b2ba2b0062e/src/instrumentation/health.go#L26 or be in one central location